### DISCUSSION TITLE
Is it possible to add @InterfaceType() to generated models?

### DISCUSSION THREAD

**Discussion by dickfickling** [MichalLytek/typegraphql-prisma#332] [Q&A]
I have a User model in my schema that I extend with an AuthorizedUser class to represent the logged-in user and expose private info.

Schema:
```
model User {
  id        String    @id @default(cuid()) @db.Text()
  firstName String    @db.Text()
  /// @TypeGraphQL.omit(output: true, input: ["where", "orderBy"])
  email     String    @unique @db.Text()
}
```

Extended User type
```
@ObjectType()
export class AuthorizedUser extends User {}
```

Authorized User Resolver:
```
@Resolver(() => AuthorizedUser)
export default class AuthorizedUserResolver {

  @Authorized()
  @Query(() => AuthorizedUser)
  authorizedUser(@CurrentUser() user: AuthorizedUser): AuthorizedUser {
    return user;
  }

  @FieldResolver(() => String)
  email(@Root() user: AuthorizedUser): string {
    return user.email!;
  }
}
```

This all seems to work, and I can query users and authorized users as expected. However, on the front end, I can't use a single fragment to query either model.

```
fragment userProfile on User {
  id
  firstName
}

query user($where: UserWhereUniqueInput!) {
  user(where: $where) {
    ...userProfile
  }
}

query fullAuthorizedUser {
  authorizedUser {
    ...userProfile
    email
  }
}
```
I get the error `Fragment "userProfile" cannot be spread here as objects of type "AuthorizedUser" can never be of type "User".`

So naturally, I'd like to add an annotation to my AuthorizedUser class that indicates that it implements `User`, a la `@ObjectType({ implements: User })`. But I also need to specify that `User` is an `InterfaceType()`, which I can't seem to do, because it's generated automatically by typegraphql-prisma.

Two questions:
- am I doing _everything_ wrong?
- if I'm not, how can I indicate to type-graphql that `User` is an `InterfaceType`?

**Comment by MichalLytek** [comment]
In the docs you will find a way to add more fields to generated models.

**Reply by dickfickling** [reply]
this is not what i’m asking about

**Reply by MichalLytek** [reply]
Instead of `class AuthorizedUser extends User` you can add more fields to `User` model and guard them with `@Authorized` or a middleware
